### PR TITLE
Layouts (stage 1): always print inferred layout on type declaration when the user asks twice

### PR DESCRIPTION
**Pull request by ncik-roberts**
If verbosity is greater than 0, then print the inferred layout on type declarations. Note: we print the *inferred* layout, not the user-written layout. You can see the difference in the `type t4 : immediate64 = A` test case.

This matches the design I sent around internally, and even uses the same test cases.

### CODE REVIEW

**Review by goldfirere — APPROVED** [review]

**Comment by ncik-roberts** [comment]
@alanechang Would you mind reviewing from scratch? Richard looked earlier but there have since been a few minor changes.

**Review by alanechang — APPROVED** [review]
Changes look good. There's just one comment that may be out-of-date.

File: src/ocaml/typing/printtyp.mli [hunk 1]
```diff
@@ -136,6 +136,19 @@ val shared_type_scheme: formatter -> type_expr -> unit
     generators; most use cases, such as error messages, have narrower contexts
     for which [type_scheme] is better suited. *)
 
+val type_declaration_for_merlin:
+  (* Like [type_declaration].
+
+     [print_non_value_inferred_jkind] is a setting controlled by merlin
+     verbosity levels. When it's true, merlin will always print
+     layout annotations (even [value]!) on type declarations.
```
[the remark this answers is not in the document]
Comment (ncik-roberts) [reply src/ocaml/typing/printtyp.mli]: Fixed!

**Comment by ncik-roberts** [comment]
@alanechang The two tests you suggested in #27 (which are good) both show the printing of `type t : value [= ...]` on the first occasion the user queries the type. This is a bit weird but is consistent with the compiler's type printer. To the extent that we want to change this, we should do so in flambda-backend (but I think it's fine to leave as-is for now).

**Comment by alanechang** [comment]
Yeah, the behavior around those tests are a bit weird, but I agree it's not a big problem and we can keep it as-is for now.

**Pull request merged**